File: Venmo/src/main/java/com/braintreepayments/api/VenmoResult.java
package com.braintreepayments.api;

public class VenmoResult {

    private VenmoAccountNonce venmoAccountNonce;
    private Exception error;
    VenmoResult(VenmoAccountNonce venmoAccountNonce) {
        this.venmoAccountNonce = venmoAccountNonce;
    }

    VenmoResult(Exception venmoError) {
        this.error = venmoError;
    }

    public VenmoAccountNonce getVenmoAccountNonce() {
        return venmoAccountNonce;
    }

    public Exception getError() {
        return error;
    }
}

File: Venmo/src/main/java/com/braintreepayments/api/venmo/VenmoResult.kt
package com.braintreepayments.api.venmo

/**
 * Result of tokenizing a Venmo account
 */
sealed class VenmoResult {

    /**
     * The Venmo flow completed successfully. This [nonce] should be sent to your server.
     */
    class Success internal constructor(val nonce: VenmoAccountNonce) : VenmoResult()

    /**
     * There was an [error] in the Venmo payment flow.
     */
    class Failure internal constructor(val error: Exception) : VenmoResult()

    /**
     * The user canceled the Venmo payment flow.
     */
    data object Cancel : VenmoResult()
}
